src/navigation/reroute.js

```diff
@@ -92,9 +105,16 @@ export function reroute(pendingPromises = [], eventArguments) {
       window.dispatchEvent(
         new CustomEvent(
           "single-spa:before-routing-event",
-          getCustomEventDetail(true)
+          getCustomEventDetail(true, { cancelNavigation })
         )
       );
+
+      if (navigationIsCanceled) {
+        finishUpAndReturn();
+        navigateToUrl(oldUrl);
```

Comment:
This part is new to my most recent commit - we need to set the browser url back to the oldUrl in order for the cancelation to really happen :)